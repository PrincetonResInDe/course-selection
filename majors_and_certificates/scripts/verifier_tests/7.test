COS-AB
2018
[
  [
    {
      "name": "COS 126",
      "title": "Computer Science: An Interdisciplinary Approach",
      "id": "1184002051",
      "dist_area": "QR"
    },
    {
      "name": "COS 226",
      "title": "Algorithms and Data Structures",
      "id": "1184002054",
      "dist_area": "QR"
    },
    {
      "name": "COS 217",
      "title": "Introduction to Programming Systems",
      "id": "1184002053",
      "dist_area": "QR"
    }
  ],
  [
    {
      "name": "COS340",
      "title": "Reasoning about Computation",
      "id": "1184010455",
      "dist_area": "QR"
    },
    {
      "name": "COS 333",
      "title": "Advanced Programming Techniques",
      "id": "1184002065",
      "dist_area": null
    }
  ],
  [
    {
      "name": "COS 351/ SOC 353 /WWS351",
      "title": "Information Technology and Public Policy",
      "id": "1184012068",
      "dist_area": "SA"
    },
    {
      "name": "ELE 206/COS 306",
      "title": "Contemporary Logic Design",
      "id": "1184022331",
      "dist_area": "STL"
    },
    {
      "name": "MUS314/COS314",
      "title": "Computer and Electronic Music",
      "id": "118421927",
      "dist_area": "QR"
    },
    {
      "name": "COS 423",
      "title": "Theory of Algorithms",
      "id": "1184002071",
      "dist_area": null
    },
    {
      "name": "COS 436",
      "title": "Human-Computer Interface Technology",
      "id": "1184021337",
      "dist_area": null
    }
  ],
  [
    {
      "name": "ART 213",
      "title": "Modernist Art: 1900 to 1950",
      "id": "1184021312",
      "dist_area": "LA"
    },
    {
      "name": "NEU 330 / PSY 330",
      "title": "Computational Modeling of Psychological Function",
      "id": "1184005628",
      "dist_area": "STL"
    },
    {
      "name": "COS 401 / LIN 304 / TRA 301",
      "title": "Introduction to Machine Translation",
      "id": "1184010983",
      "dist_area": null
    }
  ]
]