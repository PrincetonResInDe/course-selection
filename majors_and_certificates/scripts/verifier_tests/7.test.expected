[
  [
    {
      "name": "COS 126",
      "possible_reqs": [
        "Computer Science - AB//Prerequisites//Computer Science Prerequisites"
      ],
      "reqs_satisfied": [
        "Computer Science - AB//Prerequisites//Computer Science Prerequisites"
      ],
      "settled": [
        "Computer Science - AB//Prerequisites//Computer Science Prerequisites"
      ]
    },
    {
      "name": "COS 226",
      "possible_reqs": [
        "Computer Science - AB//Prerequisites//Computer Science Prerequisites"
      ],
      "reqs_satisfied": [
        "Computer Science - AB//Prerequisites//Computer Science Prerequisites"
      ],
      "settled": [
        "Computer Science - AB//Prerequisites//Computer Science Prerequisites"
      ]
    },
    {
      "name": "COS 217",
      "possible_reqs": [
        "Computer Science - AB//Prerequisites//Computer Science Prerequisites"
      ],
      "reqs_satisfied": [
        "Computer Science - AB//Prerequisites//Computer Science Prerequisites"
      ],
      "settled": [
        "Computer Science - AB//Prerequisites//Computer Science Prerequisites"
      ]
    }
  ],
  [
    {
      "name": "COS340",
      "possible_reqs": [
        "Computer Science - AB//Core Courses//Theory//000",
        "Computer Science - AB//Core Courses//General"
      ],
      "reqs_satisfied": []
    },
    {
      "name": "COS 333",
      "possible_reqs": [
        "Computer Science - AB//Core Courses//Systems//000",
        "Computer Science - AB//Core Courses//General"
      ],
      "reqs_satisfied": []
    }
  ],
  [
    {
      "name": "COS 351/ SOC 353 /WWS351",
      "possible_reqs": [
        "Computer Science - AB//Core Courses//General"
      ],
      "reqs_satisfied": [
        "Computer Science - AB//Core Courses//General"
      ],
      "settled": [
        "Computer Science - AB//Core Courses//General"
      ]
    },
    {
      "name": "ELE 206/COS 306",
      "possible_reqs": [
        "Computer Science - AB//Core Courses//Systems//000",
        "Computer Science - AB//Core Courses//General"
      ],
      "reqs_satisfied": []
    },
    {
      "name": "MUS314/COS314",
      "possible_reqs": [
        "Computer Science - AB//Core Courses//Applications",
        "Computer Science - AB//Core Courses//General"
      ],
      "reqs_satisfied": []
    },
    {
      "name": "COS 423",
      "possible_reqs": [
        "Computer Science - AB//Core Courses//Theory//000",
        "Computer Science - AB//Core Courses//General"
      ],
      "reqs_satisfied": []
    },
    {
      "name": "COS 436",
      "possible_reqs": [
        "Computer Science - AB//Core Courses//Applications",
        "Computer Science - AB//Core Courses//General"
      ],
      "reqs_satisfied": []
    }
  ],
  [
    {
      "name": "ART 213",
      "possible_reqs": [],
      "reqs_satisfied": []
    },
    {
      "name": "NEU 330 / PSY 330",
      "possible_reqs": [
        "Computer Science - AB//Core Courses//General"
      ],
      "reqs_satisfied": [
        "Computer Science - AB//Core Courses//General"
      ],
      "settled": [
        "Computer Science - AB//Core Courses//General"
      ]
    },
    {
      "name": "COS 401 / LIN 304 / TRA 301",
      "possible_reqs": [
        "Computer Science - AB//Core Courses//Applications",
        "Computer Science - AB//Core Courses//General"
      ],
      "reqs_satisfied": []
    }
  ]
]

{
  "name": "Computer Science - AB",
  "code": "COS-AB",
  "degree": "AB",
  "path_to": "Computer Science - AB",
  "urls": [
    "https://ua.princeton.edu/academic-units/department-computer-science-ab",
    "https://www.cs.princeton.edu/ugrad/becoming-cs-major/requirements"
  ],
  "contacts": [
    {
      "type": "Program Coordinator",
      "name": "Colleen Kenny",
      "email": "ckenny@cs.princeton.edu"
    },
    {
      "type": "Departmental Representative",
      "name": "Brian Kernighan",
      "email": "bwk@cs.princeton.edu"
    },
    {
      "type": "Departmental Representative",
      "name": "Andrea LaPaugh",
      "email": "aslp@cs.princeton.edu"
    }
  ],
  "satisfied": false,
  "count": 0,
  "min_needed": 2,
  "max_counted": null,
  "req_list": [
    {
      "name": "Prerequisites",
      "path_to": "Computer Science - AB//Prerequisites",
      "explanation": "The prerequisites for AB candidates are MAT 103, 104, and 202 or 204 ; COS 126; COS 217 and 226. Students should plan to take both 217 and 226 before the junior year. One or both of these are required prerequisites for all later computer science courses.",
      "pdfs_allowed": false,
      "completed_by_semester": 4,
      "satisfied": false,
      "count": 3,
      "min_needed": 6,
      "max_counted": 1,
      "req_list": [
        {
          "name": "Calculus",
          "path_to": "Computer Science - AB//Prerequisites//Calculus",
          "explanation": "COS AB candidates must also take Math 103-104 and one of Math 202 or 204.",
          "satisfied": false,
          "count": 0,
          "min_needed": 2,
          "max_counted": 2,
          "settled": [],
          "unsettled": [],
          "collapsed_course_list": [
            "MAT 103",
            "MAT 104"
          ],
          "collapsed_dist_list": []
        },
        {
          "name": "Linear Algebra",
          "path_to": "Computer Science - AB//Prerequisites//Linear Algebra",
          "explanation": "COS AB candidates must also take Math 103-104 and one of Math 202 or 204.",
          "satisfied": false,
          "count": 0,
          "min_needed": 1,
          "max_counted": 1,
          "settled": [],
          "unsettled": [],
          "collapsed_course_list": [
            "MAT 202",
            "MAT 204"
          ],
          "collapsed_dist_list": []
        },
        {
          "name": "Computer Science Prerequisites",
          "path_to": "Computer Science - AB//Prerequisites//Computer Science Prerequisites",
          "explanation": "The three prerequisite computer science courses introduce the core concepts and techniques of the field. ",
          "satisfied": true,
          "count": 3,
          "min_needed": 3,
          "max_counted": 3,
          "settled": [
            "COS 126",
            "COS 226",
            "COS 217"
          ],
          "unsettled": [],
          "collapsed_course_list": [
            "COS 126",
            "COS 217",
            "COS 226"
          ],
          "collapsed_dist_list": []
        }
      ],
      "collapsed_course_list": [
        "COS 126",
        "COS 217",
        "COS 226",
        "MAT 103",
        "MAT 104",
        "MAT 202",
        "MAT 204"
      ],
      "collapsed_dist_list": []
    },
    {
      "name": "Core Courses",
      "path_to": "Computer Science - AB//Core Courses",
      "explanation": "Eight additional departmental courses at or above the 300 level must be elected to fulfill the departmental requirements(link is external).  These eight courses must include two each from the following three areas:",
      "pdfs_allowed": false,
      "satisfied": false,
      "count": 1,
      "min_needed": 4,
      "max_counted": 1,
      "req_list": [
        {
          "name": "Theory",
          "path_to": "Computer Science - AB//Core Courses//Theory",
          "explanation": null,
          "satisfied": false,
          "count": 0,
          "min_needed": 2,
          "max_counted": 1,
          "settled": [],
          "unsettled": [
            "COS340",
            "COS 423"
          ],
          "collapsed_course_list": [
            "COS 340",
            "COS 423",
            "COS 433",
            "COS 441",
            "COS 445",
            "COS 451",
            "COS 487",
            "COS 488",
            "COS 510",
            "COS 516",
            "COS 533"
          ],
          "collapsed_dist_list": []
        },
        {
          "name": "Systems",
          "path_to": "Computer Science - AB//Core Courses//Systems",
          "explanation": null,
          "satisfied": false,
          "count": 0,
          "min_needed": 2,
          "max_counted": 1,
          "settled": [],
          "unsettled": [
            "COS 333",
            "ELE 206/COS 306"
          ],
          "collapsed_course_list": [
            "COS 306/ELE 206",
            "COS 318",
            "COS 320",
            "COS 333",
            "COS 375/ELE 375",
            "COS 418",
            "COS 425",
            "COS 461",
            "COS 463",
            "COS 475/ELE 475",
            "COS 518",
            "COS 561"
          ],
          "collapsed_dist_list": []
        },
        {
          "name": "Applications",
          "path_to": "Computer Science - AB//Core Courses//Applications",
          "explanation": null,
          "satisfied": false,
          "count": 0,
          "min_needed": 2,
          "max_counted": 1,
          "settled": [],
          "unsettled": [
            "MUS314/COS314",
            "COS 436",
            "COS 401 / LIN 304 / TRA 301"
          ],
          "collapsed_course_list": [
            "COS 314/MUS 314",
            "COS 323/ORF 363",
            "COS 324",
            "COS 325/MUS 315",
            "COS 326",
            "COS 401/TRA 301",
            "COS 402",
            "COS 424",
            "COS 426",
            "COS 429",
            "COS 432",
            "COS 435",
            "COS 436",
            "COS 455/MOL 455",
            "COS 485"
          ],
          "collapsed_dist_list": []
        },
        {
          "name": "General",
          "path_to": "Computer Science - AB//Core Courses//General",
          "explanation": "Any computer science (COS) course at the 300 or 400 level (except independent work) counts as a computer science departmental.\nAny 300- or 400- level Math or ELE or Physics or ORF course that does not duplicate COS content, Philosophy 312, MAE 345, CHM 303, ECO 312, MOL 437/NEU 437, NEU 330, and ECO 326 \u2013 Economics of the Internet count as departmentals.",
          "satisfied": true,
          "count": 2,
          "min_needed": 2,
          "max_counted": 1,
          "settled": [
            "COS 351/ SOC 353 /WWS351",
            "NEU 330 / PSY 330"
          ],
          "unsettled": [
            "COS340",
            "COS 333",
            "ELE 206/COS 306",
            "MUS314/COS314",
            "COS 423",
            "COS 436",
            "COS 401 / LIN 304 / TRA 301"
          ],
          "collapsed_course_list": [
            "CHM 303",
            "COS 3**",
            "COS 4**",
            "ECO 312",
            "ECO 326",
            "ELE 3**",
            "ELE 4**",
            "MAE 345",
            "MAT 3**",
            "MAT 4**",
            "MOL 437/NEU 437",
            "NEU 330",
            "ORF 3**",
            "ORF 4**",
            "PHI 312",
            "PHY 3**",
            "PHY 4**"
          ],
          "collapsed_dist_list": []
        }
      ],
      "collapsed_course_list": [
        "CHM 303",
        "COS 3**",
        "COS 306/ELE 206",
        "COS 314/MUS 314",
        "COS 318",
        "COS 320",
        "COS 323/ORF 363",
        "COS 324",
        "COS 325/MUS 315",
        "COS 326",
        "COS 333",
        "COS 340",
        "COS 375/ELE 375",
        "COS 4**",
        "COS 401/TRA 301",
        "COS 402",
        "COS 418",
        "COS 423",
        "COS 424",
        "COS 425",
        "COS 426",
        "COS 429",
        "COS 432",
        "COS 433",
        "COS 435",
        "COS 436",
        "COS 441",
        "COS 445",
        "COS 451",
        "COS 455/MOL 455",
        "COS 461",
        "COS 463",
        "COS 475/ELE 475",
        "COS 485",
        "COS 487",
        "COS 488",
        "COS 510",
        "COS 516",
        "COS 518",
        "COS 533",
        "COS 561",
        "ECO 312",
        "ECO 326",
        "ELE 3**",
        "ELE 4**",
        "MAE 345",
        "MAT 3**",
        "MAT 4**",
        "MOL 437/NEU 437",
        "NEU 330",
        "ORF 3**",
        "ORF 4**",
        "PHI 312",
        "PHY 3**",
        "PHY 4**"
      ],
      "collapsed_dist_list": []
    },
    {
      "name": "Independent Work",
      "path_to": "Computer Science - AB//Independent Work",
      "explanation": "All A.B. concentrators engage in independent work supervised by a member of the department. A junior project normally involves the study and solution of specific problems, often associated with a research project. It may require a significant programming effort, a theoretical study involving the design and analysis of algorithms, or an applications problem in some other field. The results of these efforts must be presented in two written reports and posters that correspond to the work undertaken in each of the terms. The senior thesis may be a study in greater depth of one of the subjects considered in junior independent work, or it may deal with another aspect of computer science and its application.",
      "satisfied": true,
      "count": 0,
      "min_needed": 0,
      "max_counted": null,
      "settled": [],
      "unsettled": [],
      "collapsed_course_list": [],
      "collapsed_dist_list": []
    }
  ],
  "collapsed_course_list": [
    "CHM 303",
    "COS 126",
    "COS 217",
    "COS 226",
    "COS 3**",
    "COS 306/ELE 206",
    "COS 314/MUS 314",
    "COS 318",
    "COS 320",
    "COS 323/ORF 363",
    "COS 324",
    "COS 325/MUS 315",
    "COS 326",
    "COS 333",
    "COS 340",
    "COS 375/ELE 375",
    "COS 4**",
    "COS 401/TRA 301",
    "COS 402",
    "COS 418",
    "COS 423",
    "COS 424",
    "COS 425",
    "COS 426",
    "COS 429",
    "COS 432",
    "COS 433",
    "COS 435",
    "COS 436",
    "COS 441",
    "COS 445",
    "COS 451",
    "COS 455/MOL 455",
    "COS 461",
    "COS 463",
    "COS 475/ELE 475",
    "COS 485",
    "COS 487",
    "COS 488",
    "COS 510",
    "COS 516",
    "COS 518",
    "COS 533",
    "COS 561",
    "ECO 312",
    "ECO 326",
    "ELE 3**",
    "ELE 4**",
    "MAE 345",
    "MAT 103",
    "MAT 104",
    "MAT 202",
    "MAT 204",
    "MAT 3**",
    "MAT 4**",
    "MOL 437/NEU 437",
    "NEU 330",
    "ORF 3**",
    "ORF 4**",
    "PHI 312",
    "PHY 3**",
    "PHY 4**"
  ],
  "collapsed_dist_list": []
}
